### ISSUE TITLE
Design the accessibility modal

### ISSUE THREAD

**Issue by mindrones** [nestauk/eurito_indicators#35]
We need to give the user the ability to open a modal where to choose various accessibility features.

Clicking on a "a11y person" icon (<img width="28" alt="Screenshot 2021-04-12 at 17 12 02" src="https://user-images.githubusercontent.com/1309648/114426698-3c1dc500-9bb2-11eb-8e8c-9047bea7ce82.png">, that we call "Leonardo") would show a modal with various multi-state-toggles that the user would repeatedly click to find their preferred setup.

The modal should easily be closed by clicking on the background like the modals we already have, also using a similar appearance.

The user might want to temporarily disable all a11y options, so in the modal we might want to have a toggle that by default shows the closed eye icon, and that when pressed disables all options and turns into the open eye icon, and so on.

To avoid the user having to click multiple times on many options to reset them all, we might also have a reset button that clears all options when pressed.

Whenever we enable one or more a11y options, the Leonardo icon would turn "on":
- "OFF" state: black stroke on white background
- "ON" state: white stroke on blue background (<img width="51" alt="Screenshot 2021-04-19 at 12 24 29" src="https://user-images.githubusercontent.com/1309648/115228878-3839f780-a10a-11eb-96f7-4935f1dc986d.png">).

If we set up some a11y options but we temporarily disable them with the eye toggle in the modal, independently from the modal visibility when should turn the Leonardo icon to the "off" state.

**Comment by mindrones** [comment]
# How to enable the a11y modal - Desktop

The A11y icon is on the top-right corner, in the header.

Clicking on it a modal appears.
Clicking on it again the modal disappears.

![MacBook - 1](https://user-images.githubusercontent.com/1309648/115715871-7d5f6300-a370-11eb-8218-b5033c1177dd.png)

![MacBook - 2](https://user-images.githubusercontent.com/1309648/115849098-768e2a00-a41c-11eb-9062-cbd720a4eded.png)

**Comment by mindrones** [comment]
# How to enable the a11y modal: Mobile - Portrait

The A11y icon is on the bottom-right corner, in the footer.

Clicking on it a modal appears.
Clicking on it again the modal disappears.

If the navigation menu is open, the modal is shown below the footer, so that it's easily accessible with the thumb

![X - 1](https://user-images.githubusercontent.com/1309648/115715920-88b28e80-a370-11eb-9b57-9dedbc9c0776.png)

![X - 2](https://user-images.githubusercontent.com/1309648/116126211-48c81000-a6be-11eb-99b3-ab9ba8daec0f.png)

![X - 3](https://user-images.githubusercontent.com/1309648/116126213-4960a680-a6be-11eb-8ca6-89d06fd65a6d.png)

![X - 4](https://user-images.githubusercontent.com/1309648/115715928-89e3bb80-a370-11eb-8f76-cf402dc0efdc.png)

**Comment by mindrones** [comment]
# Font family control

We show the current font and 4 radios to select another one.

Possibly we want to show the font name above each radio, but that can be difficult especially on mobile, so we should consider labeling them with numbers as a visual aid.

![Modal-Font-1](https://user-images.githubusercontent.com/1309648/115716775-7127d580-a371-11eb-80f9-eb585552de1a.png)

![Modal-Font-2](https://user-images.githubusercontent.com/1309648/115716777-71c06c00-a371-11eb-9028-67e4f1cb6489.png)

**Comment by mindrones** [comment]
# Font scale control

We show the current font scale as percentage and radios to select another one.

We show font scale as percentage above each radio.

![Modal-Font-scale-1](https://user-images.githubusercontent.com/1309648/115717318-fad7a300-a371-11eb-94f1-e4bbf0b1d93f.png)

![Modal-Font-scale-2](https://user-images.githubusercontent.com/1309648/115717322-fb703980-a371-11eb-8c09-a3902a2cffda.png)

**Comment by mindrones** [comment]
# Line height control

We show the current line height as percentage and radios to select another one.

We show line height as percentage above each radio.

![Modal-line-height-1](https://user-images.githubusercontent.com/1309648/115717704-56a22c00-a372-11eb-85a0-92a62d1a031f.png)

![Modal-line-height-2](https://user-images.githubusercontent.com/1309648/115717711-573ac280-a372-11eb-8e9a-b6b97a49af5b.png)

**Comment by mindrones** [comment]
# Letter spacing control

We show the current letter spacing as percentage and radios to select another one.

We show letter spacing as percentage above each radio.

![Modal-letter-spacing-1](https://user-images.githubusercontent.com/1309648/115717791-6f124680-a372-11eb-824d-496ac41d0923.png)

![Modal-letter-spacing-2](https://user-images.githubusercontent.com/1309648/115717795-6faadd00-a372-11eb-9d80-a5d9e89452d8.png)

**Comment by mindrones** [comment]
# Word spacing control

We show the current word spacing as percentage and radios to select another one.

We show word spacing as percentage above each radio.

![Modal-word-spacing-1](https://user-images.githubusercontent.com/1309648/115718006-a123a880-a372-11eb-9403-85c433779ff7.png)

![Modal-word-spacing-2](https://user-images.githubusercontent.com/1309648/115718010-a1bc3f00-a372-11eb-8543-0edb51e821d3.png)

**Comment by mindrones** [comment]
# Font width control

We show the current font width as percentage and radios to select another one.

We show font width as percentage above each radio.

![Modal-font-width-1](https://user-images.githubusercontent.com/1309648/115718269-dcbe7280-a372-11eb-9be3-00a77afefd81.png)

![Modal-font-width-2](https://user-images.githubusercontent.com/1309648/115718274-dd570900-a372-11eb-844c-069a8689b409.png)

Note that this feature depends on the selected font: if it's not available for the selected font we show a message instead:

![Modal-font-width-no-3](https://user-images.githubusercontent.com/1309648/115718275-ddef9f80-a372-11eb-8719-19e66ebb2cff.png)

# Font weight control

We show the current font weight as percentage and radios to select another one.

We show font weight as percentage above each radio.

![Modal-font-weight-1](https://user-images.githubusercontent.com/1309648/115718539-1becc380-a373-11eb-9ec0-cbb40ee4d6b4.png)

![Modal-font-weight-2](https://user-images.githubusercontent.com/1309648/115718543-1becc380-a373-11eb-83ff-f391dcfda666.png)

Note that this feature depends on the selected font: if it's not available for the selected font we show a message instead:

![Modal-font-weight-no-3](https://user-images.githubusercontent.com/1309648/115718544-1c855a00-a373-11eb-9274-7aff9f32f494.png)

**Comment by mindrones** [comment]
# Invert color control

We show the current setting of the invert boolean and a switch:

![Modal-invert-1](https://user-images.githubusercontent.com/1309648/115718821-68380380-a373-11eb-9c29-45f928e93588.png)

![Modal-invert-2](https://user-images.githubusercontent.com/1309648/115718832-69693080-a373-11eb-9d82-88c484d3db1e.png)

**Comment by mindrones** [comment]
# Brightness and contrast controls

We show the current value of brigthness / contrast as a percetage and a slider to control it.

## Brightness

![Modal-brightness-1](https://user-images.githubusercontent.com/1309648/115719206-c9f86d80-a373-11eb-96ce-e4e4d0341c30.png)

![Modal-brightness-2](https://user-images.githubusercontent.com/1309648/115719210-ca910400-a373-11eb-85ee-ba4c301a8475.png)

## Contrast

![Modal-contrast-1](https://user-images.githubusercontent.com/1309648/115719211-ca910400-a373-11eb-809a-21bd2feaf2fe.png)

![Modal-contrast-2](https://user-images.githubusercontent.com/1309648/115719212-cb299a80-a373-11eb-9db2-aff82f1c1d4f.png)

**Comment by mindrones** [comment]
# Color vision deficiency compensation control

We show the current CVD compensation and radios to select another one.

![Modal-CVD-1](https://user-images.githubusercontent.com/1309648/115719520-1479ea00-a374-11eb-989c-a7b4292778a4.png)

![Modal-CVD-2](https://user-images.githubusercontent.com/1309648/115719522-15128080-a374-11eb-940a-b4db10be7af9.png)

We would want to show names of the 3 deficiencies above each radio, but as there are pretty long we might go with their initials. In that case the current value also shows the initial to be visually consistent.

![Modal-CVD-short](https://user-images.githubusercontent.com/1309648/115719523-15128080-a374-11eb-85ea-61f4e3181242.png)

**Comment by mindrones** [comment]
# Hue shift control

We show the current hue shift in degrees and a slider to select new values.

Every 60° we show a label with the initial of the color (Red, Yellow, Green, Cyan, Blue, Magenta, Red).

![Modal-hue-shift-1](https://user-images.githubusercontent.com/1309648/115720347-e5b04380-a374-11eb-981b-40d49856d102.png)

![Modal-hue-shift-2](https://user-images.githubusercontent.com/1309648/115720351-e648da00-a374-11eb-899e-dbc60a15313a.png)

**Comment by mindrones** [comment]
# Greyscale control

We show the current value in percentage and a slider to select new values, with labels on top of its extremes.

![Modal-greyscale-1](https://user-images.githubusercontent.com/1309648/115720904-653e1280-a375-11eb-92d0-ed115d841cd3.png)

![Modal-greyscale-2](https://user-images.githubusercontent.com/1309648/115720901-64a57c00-a375-11eb-90f2-5367334439a5.png)

**Comment by mindrones** [comment]
# Navigation

## Switching control

On the right we have 2 icons to navigate to the previous and the next control.

The first control shows the "previous" icon (`ChevronLeft`) greyed out, while the last control shows the "next" icon (`ChevronRight`) greyed out. We don't cycle through the controls.

## Effect reset buttons

On the left we have 2 icons for text effects suppression and color effects suppression.
As soon as the user selects a text effect the text icon goes from grey to full black, for example here:

![Modal-Font-1](https://user-images.githubusercontent.com/1309648/115721236-c4038c00-a375-11eb-98ef-8d19a0afc744.png)

![Modal-Font-2](https://user-images.githubusercontent.com/1309648/115721239-c49c2280-a375-11eb-810d-e956b42b9717.png)

As soon as the user selects a color effect the color icon goes from grey to full black, for example here:

![Modal-invert-1](https://user-images.githubusercontent.com/1309648/115721347-dd0c3d00-a375-11eb-92e6-14fc7c528f47.png)

![Modal-invert-2](https://user-images.githubusercontent.com/1309648/115721350-dda4d380-a375-11eb-9977-3b5d095d0d75.png)

When the effects are set back to default, these 2 icons go back to greyscale.
To reset text or color effects we click on these icons. Before actually resetting the values we ask for confirmation:

![Modal-confirm-text-1](https://user-images.githubusercontent.com/1309648/115722363-d4683680-a376-11eb-8c66-20e30b1fd3e2.png)

![Modal-confirm-color-1](https://user-images.githubusercontent.com/1309648/115722356-d3cfa000-a376-11eb-89fe-3dc71fedfbe4.png)

**Comment by mindrones** [comment]
# Style

## A11y icon

When the a11y settings are different from the defaults, the icon fill goes to `lightgrey`, independently from the visibility of the modal. On the opposite, when we restore all a11y settings to their defaults the icon fill goes back to white again.

This is so that at any moment we can say if what we're seeing has the default style or not (also useful when taking screenshots of the app) and so that we know we can eventually open the a11y modal and quickly switch to the defaults.

## Modal

This modal should be all black & white to facilitate those with color vision deficiencies.

Aprt from the chevron left/right icons, available in Feather, I've used there icons from Material Design:
- https://materialdesignicons.com/icon/format-clear
- https://materialdesignicons.com/icon/human
- https://materialdesignicons.com/icon/palette-outline

The `palette-outline` one in particular should be barred like `format-clear` (we could just copy the bar from it and add it to `palette-outline`).

**Comment by mindrones** [comment]
Done, closing

**Comment by mindrones** [comment]
Added a new feature at #61
